src/modules/flow/iio/nodes.c

```diff
@@ -184,4 +184,136 @@ gyroscope_tick(struct sol_flow_node *node, void *data, uint16_t port, uint16_t c
     return -EIO;
 }
 
+struct temperature_data {
+    struct sol_iio_config config;
+    double scale;
+    double offset;
+    struct sol_flow_node *node;
```

Comment:
This is not required anymore, I see you did the same as the `gyroscope_data`, but I'd rather fix that one (as a different patch).

You can skip `struct sol_flow_node *node` from `struct temperature_data` and instead of passing a pointer to this last struct, you pass the `node` and then call `sol_flow_node_get_private_data(node)`, this will return your `struct temperature_data`. An example below.